docs/testplan/PMON-Chassis-Enhancements-test-plan.md

```diff
@@ -0,0 +1,672 @@
+    - [3.2 Test api get_supervisor_slot](#32-test-api-get_supervisor_slot)
+    - [3.3 Test api get_my_slot](#33-test-api-get_my_slot)
+    - [3.4 Test api is_modular_chassis](#34-test-api-is_modular_chassis)
+- [APIs for power consumption and supply APIs for modular chassis](#APIs-for-power-consumption-and-supply-APIs-for-modular-chassis)
+    - [4.1 Test api get_maximum_supplied_power](#41-test-api-get_maximum_supplied_power)
+    - [4.2 Test api get_status_master_led](#42-test-api-get_status_master_led)
+    - [4.3 Test api set_status_master_led](#43-test-api-set_status_master_led)
+- [APIs for thermal added for modular chassis](#APIs-for-thermal-added-for-modular-chassis)
+   - [5.1 Test api get_minimum_recorded](#51-test-api-get_minimum_recorded)
+   - [5.2 Test api get_maximum_recorded](#52-test-api-get_maximum_recorded)
+- [APIs for fan_drawer added for modular chassis](#APIs-for-fan_drawer-added-for-modular-chassis)
+   - [6.1 Test api get_maximum_consumed_power](#61-test-api-get_maximum_consumed_power)
+
+## Introduction
+
+This test plan ONLY covers the changes associated with PRs below:
+
+1. [Configure and show for platform chassis_modules #1145](https://github.com/Azure/sonic-utilities/pull/1145)
+2. [CHASSIS_STATE_DB on control-card for chassis state #395](https://github.com/Azure/sonic-swss-common/pull/395)
+3. [PSUd changes to compute power-budget for Modular chassis #104](https://github.com/Azure/sonic-platform-daemons/pull/104)
+4. [Introduce APIs for modular chassis support #124](https://github.com/Azure/sonic-platform-common/pull/124)
+5. [Common power consumption and supply APIs for modular chassis #136](https://github.com/Azure/sonic-platform-common/pull/136/files)
+6. [Thermalctld APIs for recording min and max temp #131](https://github.com/Azure/sonic-platform-common/pull/131)
+7. [Modular Chassis - Midplane monitoring APIs #148](https://github.com/Azure/sonic-platform-common/pull/148)
+8. [Modular-Chassis: Show midplane status #1267](https://github.com/Azure/sonic-utilities/pull/1267)
+
+
+### Definitions/Abbrevations
+  Term | Description
+  -----|-------------
+  VOQ  | Virtual Output Queue
+  PSU  | Power Suppy Unit
+  SFM  | Switch Fabric Card
+  
+### Test Plan 
+
+#### Debuggability
+
+The following are useful commands for validating the testcases that follow.
+
+1. Use redis cli dump config db using ``` redis-dump -d 4 -y -k "*CHASSI* ``` after shutdown line card, for example
+     ```
+         admin@sonic:~$ redis-dump -d 4 -y -k "*CHASSI*"
+                {
+                "CHASSIS_MODULE|LINE-CARD1": {
+                 "expireat": 1602657677.581144,
+                 "ttl": -0.001,
+                     "type": "hash",
+                     "value": {
+                     "admin_status": "down"
+                     }
+                      }
+                }
+     ```
+2. Use redis cli to dump state Db  ``` redis-dump -d 6 -y -k "*CHASSI* ```  
```

Comment:
Use single backticks for inline code, triple backticks for multi-line code blocks. Also suggest using full word `CHASSIS` in the command